lib/bn.js

```diff
@@ -2715,10 +2715,15 @@ BN.prototype.isubn = function isubn(num) {
 
   this.words[0] -= num;
 
-  // Carry
-  for (var i = 0; i < this.length && this.words[i] < 0; i++) {
-    this.words[i] += 0x4000000;
-    this.words[i + 1] -= 1;
+  if (this.words[1] === undefined && this.words[0] < 0) {
```

Comment:
`this.words[1]` is a total garbage (could be a number), if `length` is `1`. I would rather rely on `this.length` instead.